### DISCUSSION TITLE
VS Extension macOS support

### DISCUSSION THREAD

**Discussion by MeloHenrique** [nanoframework/Home#1389] [Ideas]
Hey everyone,

I'm thinking of extending  VS Extension to work with VS for Mac.

I'm up for taking this on and sending a PR if no one else is already on it. Can anyone confirm?

Quick tech questions:

Should the macOS stuff go in the same solution as the Windows code? Asking 'cause VS for Mac can be picky with some project types and dependencies.
Planning to keep the code DRY and share as much as possible between Windows and Mac. Any heads-up on dependencies or other gotchas?
I'll make sure to test on both platforms to catch any issues.

Thoughts?

**Comment by josesimoes** [comment]
Olá Henrique! 😃 

Considering Microsoft recent announcement about Visual Studio for Mac retirement:
https://devblogs.microsoft.com/visualstudio/visual-studio-for-mac-retirement-announcement/ do you think it's worth the investment?

The current debugger library depends on the old VS debugger engine that doesn't exist on MAC, which will raise the need to rewrite it to the new debugger engine.
Please don't get me wrong here, 😅 we're always asking for community involvement and this it. But I wanted to bring this up front, so you are aware of the immense amount of work that you have ahead, in case you choose to go forward with this.

**Reply by MeloHenrique** [reply]
Ah, totally forgot about the VS for Mac retirement—thanks for the reminder!

For small stuff, VS Code is fine, but for bigger projects, I've always leaned towards Visual Studio or Rider from JetBrains. Just a personal preference, not sure how others feel.

Originally, I was eyeing Rider for the extension. You guys already had something for Visual Studio, so I figured why not extend that to Mac, especially if some code could be reused.

I know building for Rider is a different game—Kotlin, starting from scratch, and all that—but I'm up for the challenge if it'd be a good add to the project.

What do you think?

**Reply by josesimoes** [reply]
@MeloHenrique completely forgot to get back to you about this. 🤦🏻‍♂️ Terribly sorry! 
If you're up to the challenge of writing a Rider plugin to add support for .NET nanoFramework, by all means, go for it. 😅

I'm not familiar with Rider model and how plugins work there but I suggest you investigate this before you commit to the task:
- Can you reuse the existing project system?
- Can you tap into the debugger? This requires writing a debug layer which usually requires a deep understanding on the debugger API.

 Happy to discuss more details.
